IdentityCore/src/cache/metadata/MSIDAccountMetadataCacheAccessor.m

```diff
@@ -130,38 +153,40 @@ - (BOOL)updateSignInStateForHomeAccountId:(NSString *)homeAccountId
         return NO;
     }
     
-    MSIDAccountMetadataCacheItem *accountMetadataItem;
-    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:homeAccountId clientId:clientId];
+    NSError *localError;
+    MSIDAccountMetadataCacheItem *cacheItem;
+    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWithClientId:clientId];
+    cacheItem = [_metadataCache accountMetadataCacheItemWithKey:key context:context error:&localError];
+    if (localError)
+    {
+        if (error) *error = localError;
+        return NO;
+    }
+    
+    if (!cacheItem)
+    {
+        cacheItem = [[MSIDAccountMetadataCacheItem alloc] initWithClientId:clientId];
+    }
     
     // Need to read existing account metetada if not setting as signed out
+    MSIDAccountMetadata *accountMetadata;
     if (state != MSIDAccountMetadataStateSignedOut)
     {
-        NSError *localError;
-        accountMetadataItem = [_metadataCache accountMetadataWithKey:key context:context error:&localError];
-        if (localError)
-        {
-            if (error) *error = localError;
-            return NO;
-        }
+        accountMetadata = [cacheItem accountMetadataForHomeAccountId:homeAccountId];
     }
     
-    if (!accountMetadataItem)
+    if (!accountMetadata)
     {
-        accountMetadataItem = [[MSIDAccountMetadataCacheItem alloc] initWithHomeAccountId:homeAccountId clientId:clientId];
+        accountMetadata = [[MSIDAccountMetadata alloc] initWithHomeAccountId:homeAccountId clientId:clientId];
     }
-    [accountMetadataItem updateSignInState:state];
     
-    return [_metadataCache saveAccountMetadata:accountMetadataItem
-                                           key:key
-                                       context:context error:error];
-}
-
-- (BOOL)loadAccountMetadataForClientId:(NSString *)clientId
-                               context:(id<MSIDRequestContext>)context
-                                 error:(NSError **)error
-{
-    MSIDAccountMetadataCacheKey *key = [[MSIDAccountMetadataCacheKey alloc] initWitHomeAccountId:nil clientId:clientId];
-    return [_metadataCache loadAccountMetadataForKey:key context:context error:error];
+    [accountMetadata updateSignInState:state];
+    
+    [cacheItem addAccountMetadata:accountMetadata forHomeAccountId:homeAccountId];
```

Comment:
nit: should we check for result/error here before saving?

Reply:
addressed :)